flow/environment.js

```diff
@@ -18,3 +18,7 @@ declare var __REACT_DEVTOOLS_GLOBAL_HOOK__: any; /*?{
 // temporary patches for React.Component and React.Element
 declare var ReactComponent: typeof React$Component;
 declare var ReactElement: typeof React$Element;
+
+declare class ReactSyntheticEvent extends SyntheticEvent {
```

Comment:
Can you avoid making it a global and instead create a file ReactSyntheticEvent.js (or append Type if it already exists) and whereever you use ReactSyntheticEvent do

```
import type { ReactSyntheticEvent } from 'ReactSyntheticEvent;
```